timemachine/cpp/src/kernels/k_nonbonded_pairs.cuh

```diff
@@ -154,27 +158,27 @@ void __global__ k_nonbonded_pairs(
             g_epsi += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DEPS>(-eps_grad * eps_j);
             g_epsj += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DEPS>(-eps_grad * eps_i);
 
-            real_du_dl -= sig_grad * (dsig_dl_i + dsig_dl_j);
+            real_du_dl += sign * sig_grad * (dsig_dl_i + dsig_dl_j);
```

Comment:
Difficult to verify at a glance if `accumulator -= x` --> `accumulator += sign * x` is applied everywhere it needs to be -- should `sign` also be applied to the `g_sigi` and `g_epsi` increments?

Reply:
(Additionally, might apply `result = sign * accumulator` at the end, instead of to each of the increments)

Reply:
What do you think about factoring this into an inlined helper function as in 275d787?

Reply:
> Difficult to verify at a glance if accumulator -= x --> accumulator += sign * x is applied everywhere it needs to be

Agreed. I originally checked this by comparing the implementations in [`v_nonbonded_unified`](https://github.com/proteneer/timemachine/blob/669320ece66894d4204909f9359bc4e079de50cd/timemachine/cpp/src/kernels/k_nonbonded.cuh#L399) with [`k_nonbonded_exclusions`](https://github.com/proteneer/timemachine/blob/669320ece66894d4204909f9359bc4e079de50cd/timemachine/cpp/src/kernels/k_nonbonded.cuh#L747). But I think your idea of changing the sign at the final accumulation is more clearly correct.

Reply:
Oh, I was too hasty, 275d787 doesn't actually work. I'm not exactly sure why, but I'll have to think more carefully about the order of the floating-to-fixed conversions.

Reply:
> should sign also be applied to the g_sigi and g_epsi increments?

Good call, I missed these (they were accidentally correct for the `Negated = true` case, but we're currently missing a test for `Negated = false`). Fixed in 1ae5fa3.

Reply:
>Good call, I missed these (they were accidentally correct for the Negated = true case, but we're currently missing a test for Negated = false). Fixed in 1ae5fa3.

Nice -- that looks good!

>What do you think about factoring this into an inlined helper function as in 275d787?
>idea of changing the sign at the final accumulation is more clearly correct.

Current implementation looks good! (Although I have a slight preference for flipping sign at the end rather than during accumulation (and reverting to multiplying by a sign constant rather than applying an inlined function).)

Reply:
>I have a slight preference for flipping sign at the end rather than during accumulation

Ahh, @mcwitt points out an issue I had overlooked with this suggestion: the starting values could be nonzero!